breadcord/bot.py

```diff
@@ -52,13 +52,20 @@ def __init__(self, *, tui_app: app.Breadcord | None = None, args: Namespace) ->
         data_dir = self.args.data or Path('data')
         data_dir.mkdir(exist_ok=True)
         self.data_dir = data_dir.resolve()
-        self.logs_dir = self.data_dir / 'logs'
-        self.logs_dir.mkdir(exist_ok=True)
-        self.modules_dir = self.data_dir / 'modules'
-        self.modules_dir.mkdir(exist_ok=True)
-        self.storage_dir = self.data_dir / 'storage'
-        self.storage_dir.mkdir(exist_ok=True)
-        self.settings_file = self.data_dir / 'settings.toml'
+
+        logs_dir = self.args.logs or self.data_dir / 'logs'
+        logs_dir.mkdir(exist_ok=True)
+        self.logs_dir = logs_dir.resolve()
+
+        modules_dir = self.data_dir / 'modules'
+        modules_dir.mkdir(exist_ok=True)
+        self.modules_dir = modules_dir.resolve()
+
+        storage_dir = self.args.storage or self.data_dir / 'storage'
+        storage_dir.mkdir(exist_ok=True)
+        self.storage_dir = storage_dir.resolve()
+
+        self.settings_file = (self.args.settings or self.data_dir / 'settings.toml').resolve()
```

Comment:
The attributes of `self.args` should be renamed to something clearer IMO - something like `self.args.logs_dir`, `self.args.storage_dir` and `self.args.settings_file` for consistency? This can be accomplished by setting the `dest` kwarg under `parser.add_argument()` in `breadcord/__main__.py`.

Reply:
The reasoning for the naming is to keep consistency with `args.data`. Should this be renamed as well?

Reply:
Oops, yeah - good call. That'd be good to get changed as well.